sdk/http/reference.mdx

```diff
@@ -27,99 +27,129 @@ Turso uses Bearer authentication, and requires your API token to be passed with
 
-type QueryResult = {
-  results: {
-    columns: Array<string>;
-    rows: Array<Array<Value>>;
-  };
-};
+### Interactive query
 
-type Response<T> = T | Error;
+Sometimes, it may be desirable to perform multiple operation on the same connection, in multiple rountrips. This may be the case for interactive transactions for example. This is supported by Hrana-over-http. Let's see how this is done.
 
-type BatchResponse = Array<QueryResult> | Error;
+Still on the `POST /v2/pipeline` endpoint:
+```json Body
+{
+    "requests": [
+        { "type": "execute", "stmt": { "sql": "BEGIN" } },
+    ]
+}
 ```
 
-The `Value` type represents an SQLite value. It has 4 variants:
-
-- Text: a UTF-8 encoded string
-- Integer: a 64-bit signed integer
-- Real: a 64-bits floating number
-- Blob: some binary data, encoded in base64
-- Null: the null value.
-
-#### Placeholders
+We send an `execute` request that starts a transaction (`BEGIN`), but unlike in our previous example, we don't close the connection right away. Let's see our response:
 
-If you need to use placeholders for values, you can do that:
-
-<CodeGroup>
-
-```json Positional
+```json Response
 {
-  "statements": [
-    {
-      "q": "INSERT INTO todos (text, completed) VALUES (?, ?)",
-      "params": ["Add a section for using positional args", 1]
-    }
-  ]
+    "baton": "m7lVVgEvknpf1P1irxHsHqrAqH7BLiwO4DQIAwr93PdZWGvdBNugLSokSsCZNkry",
+    "base_url": null,
+    "results": [
+        {
+            "type": "ok",
+            "response": {
+                "type": "execute",
+                "result": {
+                    "cols": [],
+                    "rows": [],
+                    "affected_row_count": 0,
+                    "last_insert_rowid": null,
+                    "replication_index": "1"
+                }
+            }
+        }
+    ]
 }
 ```
 
-```json Named
+We have successfully opened a transaction. We can see that we have received a `baton` back. This is because we haven't closed the connection. We can now use this baton to perform more queries on the same connection:
+
+
+```json Body
 {
-  "statements": [
-    {
-      "q": "INSERT INTO todos (text, completed) VALUES (:text, $completed)",
-      "params": {
-        ":text": "Add a section for using parameters",
-        "$completed": 1
-      }
-    }
-  ]
+    "baton": "m7lVVgEvknpf1P1irxHsHqrAqH7BLiwO4DQIAwr93PdZWGvdBNugLSokSsCZNkry",
+    "requests": [
+        { "type": "execute", "stmt": { "sql": "CREATE TABLE users (name)" } },
+        { "type": "execute", "stmt": { "sql": "INSERT INTO users VALUES (\"adhoc\")" } },
+        { "type": "execute", "stmt": { "sql": "COMMIT } },
+        { "type": "close" }
+    ]
 }
 ```
 
-</CodeGroup>
+We perform more queries in our transaction, commit, and then close the connection.
+
+Note that both transactions and connections have timeouts. Transaction have a 5 seconds window to complete, while connections get closed after 10 seconds of idle time.
+
+## Endpoints
```

Comment:
What do you think about moving the `pipeline` endpoint into this section, OR would you prefer to rename the sections here to be less about endpoints and more about obtaining version and checking health?

Reply:
I like option 2 better. Do you have suggestions for a section title?

Reply:
The idea of the Reference was to be less walkthrough and more "reference-y". The Quickstart could be a great place to put the current usage content you put together, and then the Reference can mention only the input/output spec.

Is that ok with you?